sdk/java/pulumi/src/test/java/com/pulumi/automation/LocalWorkspaceTest.java

```diff
@@ -3,30 +3,47 @@
 package com.pulumi.automation;
 
 import com.pulumi.Context;
-import com.pulumi.automation.ConfigValue;
-import com.pulumi.automation.LocalWorkspace;
-import com.pulumi.automation.LocalWorkspaceOptions;
-import com.pulumi.automation.OperationType;
-import com.pulumi.automation.UpdateKind;
-import com.pulumi.automation.UpdateState;
+import com.pulumi.resources.ComponentResource;
+import com.pulumi.resources.ComponentResourceOptions;
 
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.Timeout;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Random;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Consumer;
+import java.util.function.Predicate;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 @ExtendWith(LocalBackendExtension.class)
 public class LocalWorkspaceTest {
+    private static String normalizeConfigKey(String key, String projectName) {
+        var parts = key.split(":");
+        if (parts.length < 2)
+            return projectName + ":" + key;
+
+        return "";
+    }
+
+    private static String getTestSuffix() {
+        var random = new Random();
+        var result = random.nextInt(); // 31 bits, highest bit will be 0 (signed)
+        return Integer.toHexString(result); // hex representation
```

Comment:
Not sure these comments are that valuable -- if we want to talk about what the suffix looks like, I'd sooner we did Javadoc with an example personally.